### PR TITLE
System Bus access spec0.13

### PR DESCRIPTION
**Pull request by fhugodot**
hi,
i've updated the riscv-013.c to use System Bus Access using the draft spec 0.13
Would it be possible to have a look and to merge this ?
Do you want me to re-do it with the latest version?

Thank you for your comments

### CODE REVIEW

**Review by mwachs5 — COMMENTED** [review]
It seems there are a lot of changes unrelated to this PR. Was that intentional? Also, there is already a PR for System Bus Access support, does this provide different/additional functionality from that one?

**Comment by fhugodot** [comment]
to be honnest, i'm lost about how things are going...
This request was about the System Bus Access for the draft 0.13, version 0, which i believe was not existing before.
Changes are mainly in read_memory and write_memory...

File: src/target/riscv/riscv-013.c [hunk 1]
```diff
@@ -1636,32 +1712,84 @@ static int read_memory(struct target *target, target_addr_t address,
...
+		riscv_addr_t fin_addr = address + (count * size);
+		uint64_t access=0;
+
+		//ww favorise one off reading if there is an issu
+		if(count==1){  
+			for(uint32_t i=0; i<count ; i++){
+				access= dmi_read(target,DMI_SBCS);
+				dmi_write(target, DMI_SBADDRESS0, cur_addr);
+				//size/2 matching the bit access of the spec 0.13
+					,DMI_SBCS_SBACCESS,size/2);
```
Comment (timsifive) [inline src/target/riscv/riscv-013.c:1746]: What's the intent here? I was trying to clean up this PR, but this new code doesn't even compile.

Comment (fhugodot) [reply src/target/riscv/riscv-013.c:1746]: :(....
access=set_field(access,DMI_SBCS_SBACCESS,size/2);

Comment (fhugodot) [reply src/target/riscv/riscv-013.c:1746]: sorry...

**Comment by timsifive** [comment]
@fhugodot, I've cleaned up this PR (a lot). My try is at https://github.com/riscv/riscv-openocd/tree/old_bus2

Can you take a look and confirm that code works with your hardware?

**Comment by fhugodot** [comment]
I try today

**Comment by fhugodot** [comment]
Hi tim
Got issue when doing bootstrap:

+ aclocal
+ libtoolize --automake --copy
+ autoconf
+ autoheader
+ automake --gnu --add-missing --copy
configure.ac:28: installing './compile'
configure.ac:37: installing './config.guess'
configure.ac:37: installing './config.sub'
configure.ac:16: installing './install-sh'
configure.ac:16: installing './missing'
src/Makefile.am:4: error: bad characters in variable name '%C%_openocd_SOURCES'
Makefile.am:143:   'src/Makefile.am' included from here
src/Makefile.am:7: error: bad characters in variable name '%C%_libopenocd_la_SOURCES'
Makefile.am:143:   'src/Makefile.am' included from here
src/Makefile.am:11: error: bad characters in variable name '%C%_openocd_LDADD'
Makefile.am:143:   'src/Makefile.am' included from here
src/Makefile.am:13: error: bad characters in variable name '%C%_openocd_LDADD'
Makefile.am:143:   'src/Makefile.am' included from here
src/Makefile.am:16: error: bad characters in variable name '%C%_openocd_LDADD'
Makefile.am:143:   'src/Makefile.am' included from here
src/Makefile.am:18: error: bad characters in variable name '%C%_openocd_LDADD'
Makefile.am:143:   'src/Makefile.am' included from here
src/Makefile.am:21: error: bad characters in variable name '%C%_libopenocd_la_CPPFLAGS'
Makefile.am:143:   'src/Makefile.am' included from here
src/Makefile.am:26: error: bad characters in variable name '%C%_libopenocd_la_CPPFLAGS'
Makefile.am:143:   'src/Makefile.am' included from here
src/Makefile.am:27: error: bad characters in variable name '%C%_libopenocd_la_CPPFLAGS'
Makefile.am:143:   'src/Makefile.am' included from here
src/Makefile.am:29: error: bad characters in variable name '%C%_libopenocd_la_CPPFLAGS'
Makefile.am:143:   'src/Makefile.am' included from here
src/Makefile.am:30: error: bad characters in variable name '%C%_libopenocd_la_CPPFLAGS'
Makefile.am:143:   'src/Makefile.am' included from here
src/Makefile.am:31: error: bad characters in variable name '%C%_libopenocd_la_CPPFLAGS'
Makefile.am:143:   'src/Makefile.am' included from here
src/Makefile.am:35: error: bad characters in variable name '%C%_libopenocd_la_CPPFLAGS'
Makefile.am:143:   'src/Makefile.am' included from here
src/Makefile.am:38: error: bad characters in variable name '%C%_libopenocd_la_LDFLAGS'
Makefile.am:143:   'src/Makefile.am' included from here
src/Makefile.am:46: error: bad characters in variable name '%C%_libopenocd_la_LIBADD'
Makefile.am:143:   'src/Makefile.am' included from here
doc/Makefile.am:2: error: bad characters in variable name '%C%_openocd_TEXINFOS'
Makefile.am:144:   'doc/Makefile.am' included from here
Makefile.am:46: warning: wildcard $(srcdir: non-POSIX variable name
Makefile.am:46: (probably a GNU make extension)
Makefile.am: installing './INSTALL'
Makefile.am: installing './depcomp'
automake: error: cannot open < ./%D%/openocd.texi: No such file or directory


I’ve update automake to 1.14 and still the issue

I’ll continue to see the issue tomorrow

Regards,
Fabien

**Comment by timsifive** [comment]
This seems to be a common issue with older Linux distributions, but I'm not
sure what the real solution is. Presumably some combination of updating
something, and then doing a completely clean build.

**Comment by fhugodot** [comment]
Hi Tim,
It is happening with CentOS updated to the latest libraries.
I’ll have a look
Regards,
Fabien

**Comment by fhugodot** [comment]
Hi,
Just to share the information.
To make bootstrap working on Centos latest, run sudo yum install glib-devel xorg-x11-devel
Thanks
Fabien

**Comment by fhugodot** [comment]
Hi Tim,
I think I should report that the kind of error I’ve got when I do ./bootstrap now though:
+ aclocal
+ libtoolize --automake --copy
+ autoconf
configure.ac:11: error: possibly undefined macro: AC_MSG_WARN
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure.ac:35: error: possibly undefined macro: AC_DISABLE_SHARED
configure.ac:36: error: possibly undefined macro: AC_PROG_LIBTOOL
configure.ac:203: error: possibly undefined macro: AC_DEFINE
configure.ac:607: error: possibly undefined macro: AC_MSG_NOTICE

I didn’t get that before…
Anyway
I keep looking
Regards,
fabien

**Comment by fhugodot** [comment]
Hi Tim,
This is a final solution if people have difficulties with centOS 7.x (if this can help someone)
After downloading the branch


-          Type autoupdate configure.ac

-          On many distros, including CentOS7, the seven macro files for libtools are installed under /usr/share/aclocal rather than under /usr/local/share/aclocal. Consequently, they aren't found when the package you are building sets prefix to /usr/local.

To fix this if you already have a /usr/local/share/aclocal directory, type the following as root:

for file in argz libtool ltdl ltoptions ltsugar ltversion lt~obsolete

do

ln -s /usr/share/aclocal/$file.m4 /usr/local/share/aclocal/$file.m4

done

To fix this if you don't have a /usr/local/share/aclocal directory, type the following as root:

ln -s /usr/share/aclocal /usr/local/share/aclocal



-          Type export ACLOCAL_PATH=$HOME/install/libtool/share/aclocal:$ACLOCAL_PATH


-          You can type ./bootstrap

Regards,
Fabien

From: Fabien Hugodot
Sent: 06 March 2018 14:00
To: 'riscv/riscv-openocd' <reply@reply.github.com>; riscv/riscv-openocd <riscv-openocd@noreply.github.com>
Cc: Mention <mention@noreply.github.com>
Subject: RE: [riscv/riscv-openocd] System Bus access spec0.13 (#214)

Hi Tim,
I think I should report that the kind of error I’ve got when I do ./bootstrap now though:
+ aclocal
+ libtoolize --automake --copy
+ autoconf
configure.ac:11: error: possibly undefined macro: AC_MSG_WARN
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure.ac:35: error: possibly undefined macro: AC_DISABLE_SHARED
configure.ac:36: error: possibly undefined macro: AC_PROG_LIBTOOL
configure.ac:203: error: possibly undefined macro: AC_DEFINE
configure.ac:607: error: possibly undefined macro: AC_MSG_NOTICE

I didn’t get that before…
Anyway
I keep looking
Regards,
fabien

**Comment by timsifive** [comment]
Thanks. I'm marking this e-mail so I can easily find it the next time
reports this problem.

Tim

**Comment by fhugodot** [comment]
Hi Tim,

To make make OpenOCD working on our hardware I had to comment this parts in the examine method:

//uint32_t max_hartsel_mask = ((1L<<10)-1) << DMI_DMCONTROL_HARTSEL_OFFSET;
                //dmi_write(target, DMI_DMCONTROL, max_hartsel_mask | DMI_DMCONTROL_DMACTIVE);
                uint32_t dmcontrol = dmi_read(target, DMI_DMCONTROL);

                if (!get_field(dmcontrol, DMI_DMCONTROL_DMACTIVE)) {
                                LOG_ERROR("Debug Module did not become active. dmcontrol=0x%x",
                                                                dmcontrol);
                                return ERROR_FAIL;
                }

                //uint32_t hartsel = get_field(dmcontrol, max_hartsel_mask);
                //info->hartsellen = 0;
                //while (hartsel & 1) {
        //    info->hartsellen++;
                //            hartsel >>= 1;
                //}

I don’t quite understand what is the purpose of these bits.
We have only one hart.

Do I miss something?

Otherwise, once commented, everything works fine !

Thank you
Fabien

**Comment by timsifive** [comment]
This PR is superseded by #225.

**Pull request closed without merging**